packages/wallet/api/src/lib-wallet.js

```diff
@@ -556,11 +557,42 @@ export function makeWallet({
       });
   }
 
+  /**
+   * @param { () => ( Promise | undefined ) } get - The function whose return value
+   * to memoize.
+   */
+  const makeMemoizedGetter = get => {
+    let pk;
```

Comment:
Couldn't this be a constant? There's one assignment and I don't see the benefit to deferring it.

Reply:
I was using `pk === undefined` as a way to signal that `get()` was not in flight already. Maybe there's a better way to do so?

Reply:
Hmm, maybe I don't understand the intent of this function. Is is to defer calling the getter? Does this also do what you want?
```js
  const makeMemoizedGetter = get => {
    return () => {
      const pk = makePromiseKit();
      pk.resolve(get());
      return pk.promise;
    };
  };
```

If so, that can be simplified,
```js
  const makeMemoizedGetter = get => {
    return () => Promise.resolve(get());
  };
```
Though I wouldn't call those "memoizing" because there's no cache. It's just a deferrred promise.

Reply:
Yes it's to:
1) Defer calling the getter until its result needs to be read.
2) Only call the getter once no matter how many subsequent/concurrent reads are made for the getter's return value.

The second part works as currently written, but not with those examples.

Reply:
ha yeah, I lost that along the way. Thanks for spelling it out. A _deferred and memoized_  getter.

This reads clearer to me as a memoization idiom and doesn't require Agoric's custom promise kit:
```
const makeMemoizedGetter = get => {
  let valP;
  return () => {
    if (!result) {
      valP = Promise.resolve(get());
    }
    return valP;
  };
};
```
With types:
```
/**
 * @template T
 * @param {() => T} get
 */
const makeMemoizedGetter = get => {
  /** @type {Promise<T>?} */
  let result;
  return () => {
    if (!result) {
      result = Promise.resolve(get());
    }
    return result;
  };
};
```
not blocking